src/app/store/courses/courses.selectors.ts

```diff
@@ -1 +1,47 @@
-// Add your code here
+import { createSelector, createFeatureSelector } from "@ngrx/store";
+
+import { coursesFeatureKey } from "./courses.reducer";
+import { CoursesState } from "./courses.reducer";
+
+export interface AppState {
+  [coursesFeatureKey]: CoursesState;
```

Comment:
you could put the constant in this class, also I think the constant should be uppercase to distinguish it from other variables

Reply:
ah, I see that this is from the example from the theory